zk-chat-server-lib/src/semaphore/index.ts

```diff
@@ -35,63 +35,78 @@ class SemaphoreSynchronizer {
      * Sync commitments on startup, and schedule a job to sync on a regular interval.
      */
     public sync = async() => {
+        console.log("!@# src/semaphore/index.ts::sync: this.config");
         await this.syncCommitmentsFromSemaphore();
         await this.continuousSync();
     }
 
     public syncCommitmentsFromSemaphore = async() => {
         // On startup
-        // 1. Get all groups from interrep
-        const allGroupsOnNet: IInterRepGroupV2[] = await semaphoreFunctions.getAllGroups(this.config.interepUrl);
+        // 1. Get all groups from Semaphore
+        const allGroupsOnNet: ISemaphoreRepGroupV2[] = await semaphoreFunctions.getAllGroups(this.config.interepUrl);
         const groupsInDb: IGroup[] = await this.groupService.getGroups();
 
         let tree_root_changed = false;
 
         // 2. For each group, check the status in database. Only load new members for group if the size in db is different than the new size of the group
         for (let g of allGroupsOnNet) {
-            const g_id = g.provider + "_" + g.name;
-            const groupInDb: IGroup | undefined = groupsInDb.find(x => x.group_id == g_id && x.name == g.name && x.provider == g.provider);
-
+            const groupMembers = await this.loadGroupMembers(g.id);
+            /*
+                NICO's OBSERVATION: In the Zuzalu case, size = numberOfLeaves because the members are presented as a list.
+                I think that the inner elements of Semaphore (RLN) do not delete members from the array but they just replace them with a zero value.
+                Therefore size = active members and numberOfLeaves = total members.
+                Regarding Zuzalu, I am not sure if size is variable.
+            */
+            // TODO: get the number of leaves from the group api
+            // TODO: get the size of the leaves from the group api
+            const numberOfLeaves = groupMembers.length;
+
+            const groupInDb: IGroup | undefined = groupsInDb.find(x => x.group_id == g.id);
```

Comment:
Should we also compare the name and provider like they originally did?
```suggestion
            const groupInDb: IGroup | undefined = groupsInDb.find(x => x.group_id == g_id && x.name == g.name && x.provider == g.provider);
```

Reply:
This feature would work if we had different providers (e.g. Zuzalu and Interrep) but in this particular case, Zuzalu only returns the members list, there is no provider attribute in it